### PR TITLE
update glossary

### PR DESCRIPTION
**Pull request by austinlparker**
contributes to #21

### CODE REVIEW

File: 000 - glossary.md [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
-systems – more general definitions also exist for more general contexts but are
-not provided here.
+this project.
 
 ---
 
 _Distributed System_: A system that's characterized by its components
 being split up across multiple distinct computers on a network.
 
+_Distributed Tracing_: A method of profiling and monitoring distributed system
```
Comment (bhs) [inline 000 - glossary.md:11]: When I define DT I am pretty careful to describe how this form of tracing follows transactions, and particularly follows them across process and machine/infra/network boundaries. My quick take:

```
_Distributed Tracing_: The technique of following – or "tracing" – requests as they traverse distributed systems, representing them as telemetry, and reassembling and analyzing models of the distributed requests.
```

File: 000 - glossary.md [hunk 2]
```diff
@@ -1,39 +1,56 @@
...
+performance by capturing telemetry data associated with each request.
+
 _Instrument_: A logical object that emits a specific piece of named telemetry.
 For example, an instrument might measure the current number of requests being
 handled by a process, or the amount of free disk space.
 
 _Instrumentation_: Code that exists to emit telemetry signals to some external
 system or endpoint. For example, a logging statement or a metric instrument.
 
+_Logs_: Human-readable text output from services and computers in a system.
```
Comment (bhs) [inline 000 - glossary.md:21]: I think of Logs as  being both structured and unstructured (the "superclass", as it were)... i.e., they aren't necessarily human-readable text output.

File: 000 - glossary.md [hunk 3]
```diff
@@ -1,39 +1,56 @@
...
+
+_Metrics_: Lightweight statistical data about a system designed to be plotted
```
Comment (bhs) [inline 000 - glossary.md:23]: We should remove "Lightweight", as high-cardinality metrics are actually quite heavy...

Maybe just "statistics about resources and transactions which are gathered periodically, typically to be stored in a time series database and queried as part of a broader observability solution.

File: 000 - glossary.md [hunk 4]
```diff
@@ -1,39 +1,56 @@
...
+and tracked over time.
+
+_Monitoring_: The process of observing a system for changes.
```
Comment (bhs) [inline 000 - glossary.md:26]: "The aspect of observability that tracks the health of individual resources and components relative to the health of the larger application."

File: 000 - glossary.md [hunk 5]
```diff
@@ -1,39 +1,56 @@
...
+
 _Observability_: The ability to both understand and reason about the behavior
 of a software system without needing to make **changes** to that software system.
 In practice, observability requires both high-quality telemetry and
 high-quality observability tooling.
 
+_Resource_: A durable unit of computing that is consumed to power a transaction.
```
Comment (bhs) [inline 000 - glossary.md:33]: "A finite and durable unit of computation, storage, or other work that is (typically) shared across many distinct transactions. See … … …"

File: 000 - glossary.md [hunk 6]
```diff
@@ -1,39 +1,56 @@
...
+See [Resources](./02%20-%20customers-and-engineers.md#resources) for more
+information.
+
+_Service_: A logical process that does some useful work in a system.
```
Comment (bhs) [inline 000 - glossary.md:37]: I think we'll need to do better on this one... it's unfortunately overloaded, probably even within this project. :-/  Maybe we should stick with "microservice" to remove ambiguity with RPC services and others?

File: 000 - glossary.md [hunk 7]
```diff
@@ -1,39 +1,56 @@
...
-- Metrics
-- Logs
-- Structured Logging
-- Resource
-- Transaction
-- Monitoring
-- SLO
-- SLI
---->
+_Transaction_: A single unit of useful work in a system. See
```
Comment (bhs) [inline 000 - glossary.md:54]: "The input, output, and path of a single client's request in a cloud-native application, consuming resources along the way. See … … …"

**Review by bhs — COMMENTED** [review]
I tried to make suggestions for alternatives... "Service" is a tough one – may want to scope it to "microservice" to make it easier. :-/

File: 000 - glossary.md (hunk 6 above)
Comment (austinlparker) [reply 000 - glossary.md]: Let me know what you think about the updated text.

File: 000 - glossary.md [hunk 8]
```diff
@@ -1,39 +1,65 @@
...
 high-quality observability tooling.
 
+_Resource_: A finite and durable unit of computation, storage, or other work
+that is typically shared across many distinct transactions.
+See [Resources](./02%20-%20customers-and-engineers.md#resources) for more
+information.
+
+_Microservice_: A service in a distributed system that generally handles a
+well defined and tightly scoped amount of work. These services are generally
+designed to be available over a network link. An example would be an
```
Comment (bhs) [inline 000 - glossary.md:44]: nit: I would can the "These services ... link" sentence – that's covered by "distributed system."

**Review by bhs — APPROVED** [review]

**Pull request merged**